1-development/list_posts_by_rating/src/test/java/com/mixfar/posts/PostRestControllerTests.java

```diff
@@ -0,0 +1,141 @@
+package com.mixfar.posts;
+
+import com.mixfar.posts.controllers.PostController;
+import com.mixfar.posts.model.Post;
+import com.mixfar.posts.repositories.PostRepository;
+import com.mixfar.posts.service.PostService;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+
+import java.time.LocalDate;
+
+import static org.assertj.core.internal.bytebuddy.matcher.ElementMatchers.is;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(PostController.class)
+public class PostRestControllerTests {
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private PostService postService;
+
+    @MockBean
+    private PostRepository postRepository;
+
+    /**
+     * Create post with success
+     * @throws Exception
+     */
+    @Test
+    public void create_OK() throws Exception {
+
+        String post_json = "{\"postTitle\": \"Post created now\",\"postBody\": \"Another Post body\",\"upVotes\": 0,\"downVotes\": 0,\"rating\": 0.0,\"publishDate\": \"2021-01-24\"}";
+
+        mvc.perform(post("/create")
+                .contentType(MediaType.APPLICATION_JSON).content(post_json))
+                .andExpect(status().isCreated());
+    }
+
+    /**
+     * Get Posts By Rating Desc with Success
+     * @throws Exception
+     */
+    @Test
+    public void getPostsByRatingDesc_OK() throws Exception {
```

Comment:
This endpoint returns a list of values. Is there any test for that?

Reply:
Yes, as I do in the `upvote_NOK` test, I can check if the response has a list with posts